Tests/Services/NfieldSurveyGroupServiceTests.cs

```diff
@@ -0,0 +1,232 @@
+
+            var result = await target.CreateAsync(new SurveyGroupValues
+            {
+                Name = "Default",
+                Description = null
+            });
+
+            Assert.NotNull(result);
+            Assert.Equal("Default", result.Name);
+            Assert.Equal(null, result.Description);
+            Assert.Equal(new DateTime(1799, 11, 10), result.CreationDate);
+            Assert.Equal(1, result.SurveyGroupId);
+        }
+
+        [Fact]
+        public async Task CanCreateModifyAndDeleteGroup()
+        {
+            var createdGroup = new SurveyGroup
+            {
+                SurveyGroupId = 2,
+                Name = "Default",
+                Description = null,
+                CreationDate = new DateTime(1799, 11, 10)
+            };
+
+            var updatedGroup = new SurveyGroup
+            {
+                SurveyGroupId = 2,
+                Name = "Default with a twist",
+                Description = "Modified description",
+                CreationDate = new DateTime(1799, 11, 10)
+            };
+
+            var mockedNfieldConnection = new Mock<INfieldConnectionClient>();
+            var mockedHttpClient = CreateHttpClientMock(mockedNfieldConnection);
+            mockedHttpClient
+                .Setup(client => client.PostAsJsonAsync(new Uri(ServiceAddress, "SurveyGroups"), It.IsAny<SurveyGroupValues>()))
+                .Returns(CreateTask(HttpStatusCode.OK, new StringContent(JsonConvert.SerializeObject(createdGroup))));
+            mockedHttpClient
+                .Setup(client => client.PatchAsJsonAsync(new Uri(ServiceAddress, "SurveyGroups/2"), It.IsAny<SurveyGroupValues>()))
+                .Returns(CreateTask(HttpStatusCode.OK, new StringContent(JsonConvert.SerializeObject(updatedGroup))));
+            mockedHttpClient
+                .Setup(client => client.DeleteAsync(new Uri(ServiceAddress, "SurveyGroups/2")))
+                .Returns(CreateTask(HttpStatusCode.OK));
+
+            var target = new NfieldSurveyGroupService();
+            target.InitializeNfieldConnection(mockedNfieldConnection.Object);
+
+            var result = await target.CreateAsync(new SurveyGroupValues
+            {
+                Name = "Default",
+                Description = null
+            });
+
+            Assert.NotNull(result);
+            Assert.Equal("Default", result.Name);
+            Assert.Equal(null, result.Description);
+            Assert.Equal(new DateTime(1799, 11, 10), result.CreationDate);
+            Assert.Equal(2, result.SurveyGroupId);
+
+            // modify survey group object we got back, then post it
+            result.Name = "Default with a twist";
+            result.Description = "Modified description";
+
+            result = await target.UpdateAsync(result.SurveyGroupId, result);
+
+            Assert.NotNull(result);
+            Assert.Equal("Default with a twist", result.Name);
```

Comment:
If you expect the _result.Name_ to be the same as second group SurveyGroup Name, you can use that object property value instead of a literal, e.g

```suggestion
           Assert.Equal(expectedGroups.Where(g => g.SurveyGroupId == 2).Name ,  result.Name);
```

This would make the test more robust IMO, as you base your assert condition on a test arrange value, not input literal